src/include/cell_inline.h

```diff
@@ -662,7 +662,7 @@ __wt_cell_leaf_value_parse(WT_PAGE *page, WT_CELL *cell)
  * case, make sure all reads are inside the page image. If an error occurs, return an error code but
  * don't output messages, our caller handles that.
  */
-#define WT_CELL_LEN_CHK(t, len)                                                             \
+#define WT_CELL_LEN_CHK(t, len, dsk, end)                                                   \
     do {                                                                                    \
         if (end != NULL &&                                                                  \
           ((uint8_t *)(t) < (uint8_t *)dsk || (((uint8_t *)(t)) + (len)) > (uint8_t *)end)) \
```

Comment:
This has already gone through review - but with a switch to passing the `dsk` and `end` as parameters to the length check macro, the implementation should also wrap usage of those in brackets.

It's not likely here, but arbitrary expressions can be passed to macros, so it's necessary to wrap encapsulate them. For example if the caller of this macro did `WT_CELL_LEN_CHECK(t, 100, prior_disk + offset, prior_disk + offset + 300);` the casting in the implementation here would be wrong.

It wasn't necessary before, since the variables are just local variables, so couldn't be complex expressions that needed boundaries.

Reply:
@agorrod that's a good point, and one I should have picked up.